src/test/java/com/amazonaws/services/dynamodbv2/GetLocksByPartitionKeyTest.java

```diff
@@ -0,0 +1,175 @@
+    assertEquals(secondLock.getPartitionKey(), retrievedLock.getPartitionKey());
+    assertEquals(secondLock.getSortKey(), retrievedLock.getSortKey());
+    assertTrue(Arrays.equals(getBytes(secondLock.getData().get()), getBytes(retrievedLock.getData().get())));
+    assertEquals(secondLock.getRecordVersionNumber(), retrievedLock.getRecordVersionNumber());
+
+    firstLock.close();
+
+    allLocksFromDynamoDB = this.lockClientForRangeKeyTable.getLocksByPartitionKey(partitionKey, deleteOnRelease).collect(toList());
+
+    assertEquals(1, allLocksFromDynamoDB.size());
+    retrievedLock = allLocksFromDynamoDB.get(0);
+    assertEquals(secondLock.getPartitionKey(), retrievedLock.getPartitionKey());
+    assertEquals(secondLock.getSortKey(), retrievedLock.getSortKey());
+    assertTrue(Arrays.equals(getBytes(secondLock.getData().get()), getBytes(retrievedLock.getData().get())));
+    assertEquals(secondLock.getRecordVersionNumber(), retrievedLock.getRecordVersionNumber());
+
+    secondLock.close();
+
+    allLocksFromDynamoDB = this.lockClientForRangeKeyTable.getLocksByPartitionKey(partitionKey, deleteOnRelease).collect(toList());
+    assertEquals(Collections.emptyList(), allLocksFromDynamoDB);
+  }
+
+  @Test
+  public void getLocksFromDynamoDB_whenMultiplePages_shouldReturnMultiplePages() throws LockNotGrantedException, InterruptedException, IOException {
+    // Scan is paginated.
+    // Scans with items that are larger than 1MB are paginated
+    // and must be retrieved by performing multiple scans.
+    // Make sure the client handles pagination correctly.
+    // See
+    // http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/QueryAndScan.html#Pagination
+    final String partitionKey = "Test 1";
+    int numBytesOfData = 0;
+    final byte[] data = new byte[(DYNAMODB_MAX_ITEM_SIZE_IN_BYTES * 9) / 10];
+    final long randomSeed = System.currentTimeMillis();
+    System.out.println("Random seed is " + randomSeed);
+    final Map<String, LockItem> acquiredLockItemsByKey = new HashMap<>();
+    while (numBytesOfData < DYNAMODB_MAX_PAGE_SIZE_IN_BYTES) {
+      SECURE_RANDOM.nextBytes(data);
+      int size = acquiredLockItemsByKey.size();
+      final AcquireLockOptions options =
+          AcquireLockOptions.builder(partitionKey)
+              .withSortKey(Integer.toString(size))
+              .withData(ByteBuffer.wrap(data)).withDeleteLockOnRelease(true).build();
+      final LockItem acquiredLock = this.lockClientForRangeKeyTable.acquireLock(options);
+
+      acquiredLockItemsByKey.put(getLockKey(acquiredLock), acquiredLock);
+      numBytesOfData += acquiredLock.getData().get().array().length;
+    }
+
+    final boolean deleteOnRelease = false;
+    List<LockItem> locksWithPartitionKey = this.lockClientForRangeKeyTable.getLocksByPartitionKey(partitionKey, deleteOnRelease).collect(toList());
+
+    assertEquals(acquiredLockItemsByKey.size(), locksWithPartitionKey.size());
```

Comment:
Is there a way we can verify there were actually paginated calls to DynamoDB?

Reply:
Not as far as I know. Renamed the test case instead to check for complete results.